ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ActiveContractsServiceIT.scala

```diff
@@ -699,7 +719,21 @@ class ActiveContractsServiceIT extends LedgerTestSuite {
       _ <- ledger.create(party, Dummy(party))
       offset2 <- ledger.currentEnd()
       _ <- ledger.create(party, Dummy(party))
-      _ <- ledger.prune(pruneUpTo = offset2)
+      _ <- FutureAssertions.succeedsEventually(
+        retryDelay = 100.millis,
+        maxRetryDuration = 10.seconds,
+        ledger.delayMechanism,
+        "Pruning",
+      ) {
+        // We are retrying a command submission + pruning to make this test compatible with Canton.
+        // That's because in Canton pruning will fail unless ACS commitments have been exchanged between participants.
+        // To this end, submitting a command is prompting Canton to exchange ACS commitments
+        // and allows the pruning call to eventually succeed.
+        for {
+          _ <- ledger.submitAndWait(ledger.submitAndWaitRequest(party, Dummy(party).create.command))
+          _ <- ledger.prune(pruneUpTo = offset2, attempts = 1)
```

Comment:
Yes this is factually correct. To avoid the test flaking, you have done the right thing by repeatedly submitting commands before each prune call. I would consider tweaking the comment according to the suggestion above.